homeassistant/components/binary_sensor/tod.py

```diff
@@ -0,0 +1,250 @@
+"""
+Support for showing the binary sensors represending current time of the day.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.tod/
+
+"""
+from datetime import datetime, timedelta
+import logging
+import pytz
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, ENTITY_ID_FORMAT, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_FRIENDLY_NAME, CONF_AFTER, CONF_BEFORE,
+    CONF_SENSORS, SUN_EVENT_SUNRISE, SUN_EVENT_SUNSET)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import async_generate_entity_id
+from homeassistant.helpers.event import async_track_point_in_utc_time
+from homeassistant.helpers.sun import (
+    get_astral_event_date, get_astral_event_next)
+from homeassistant.util import dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_AFTER_OFFSET = 'after_offset'
+CONF_BEFORE_OFFSET = 'before_offset'
+
+ATTR_NEXT_UPDATE = 'next_update'
+
+
+SENSOR_SCHEMA = vol.Schema({
+    vol.Required(CONF_AFTER): vol.Any(cv.time, vol.All(
+        vol.Lower, cv.sun_event)),
+    vol.Optional(CONF_AFTER_OFFSET, default=timedelta(0)): cv.time_period,
+    vol.Required(CONF_BEFORE): vol.Any(cv.time, vol.All(
+        vol.Lower, cv.sun_event)),
+    vol.Optional(CONF_BEFORE_OFFSET, default=timedelta(0)): cv.time_period,
+    vol.Optional(ATTR_FRIENDLY_NAME): cv.string,
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
```

Comment:
You need to use the `CONF_` constants, so CONF_NAME, CONF_ENTITY_ID

Reply:
Replaced ATTR_ with CONF_